### DISCUSSION TITLE
Built in Debounce

### DISCUSSION THREAD

**Discussion by cubecleveland** [TanStack/query#3077] [Q&A] [answered]
Hello,

Is there an inbuilt debounce to useQuery lib?

We have cases in the ui wheere we are calling a  utation on a clikc of a button to togle a state in the DB and we want to debounce this.

**Comment by TkDodo** [comment, marked as answer]
no, there is not. if its a mutation, you need to debounce the call to the mutateFn for yourself, e.g. via lodash.debounce
